test/index.js

```diff
@@ -40,5 +42,21 @@ const support = require('..')
     })
   })
 
+  const cliTestsDir = path.join(__dirname, 'cli')
+  const cliTests = await fs.readdir(cliTestsDir)
+  suite('cli tests', async () => {
+    const supportCommand = '../../../bin/support'
+    cliTests.forEach((testDir) => {
+      test(`cli test: ${testDir}`, () => {
+        const cwd = path.join(__dirname, 'cli', testDir)
+        const rawCommand = fs.readFileSync(path.join(cwd, 'command')).toString()
+        const command = rawCommand.replace(/\$\{cwd\}/g, cwd)
+        const expected = fs.readFileSync(path.join(cwd, 'expected')).toString()
+        const result = childProcess.execSync(`${supportCommand} ${command}`, { cwd: cwd }).toString()
+        assert.strictEqual(result, expected)
+      })
+    })
+  })
```

Comment:
I think I see it together now, but I think disagree with the approach of making the code abstract and having duplicated files. If the abstraction did end up reducing duplication or the code complexity, then I could see adding all this but IMO this just made the PR much more difficult to review.

TBQH, I might prefer to see one object with all of this in the test file and the abstraction on the bottom than having a bunch of files with just the string `validate` in them.  IDK, I am sure this is a matter of opinion and others probably disagree with me 😛.  I am sure there is a middle ground that would also put me more at ease with this approach.

Reply:
Each test has a file for

- command to run
- expected output
- other required content like package.json package-support.json  required for the test.

In this case the file with `validate` is the command to be run for the test, in some cases there are additional options so they are not all the same.

So your suggestion is one file with multiple sections versus multiple files?  That will be more work to manage since right now its easy to generate the output file by just running the command.  Merging them into 1 will require having to merge, dealing with any required escaping etc.

Reply:
I think what really thew me was the `command` files.  I was just scrolling and scrolling this PR and seeing the same file repeated.  I do understand having the content as fixtures in a directory, as that is really a requirement.  I don't think this would be something I would ever block, but it all seems more difficult to grok to me. 🤷

Reply:
I'll do a pass through all of the other comments and fix up and then we can loop back to this one. I'd prefer to not block on this but we can update/change the tests to use a different structure if we find a better way in a later PR.

Reply:
my comment was only a reminder not a blocker btw :+1:

Reply:
Agreed, this is not a blocker.  Once we get this all out and in use we can just open an issue to get on the same page for test strategy.